orttraining/orttraining/lazy_tensor/cuda_tool.h

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifdef USE_CUDA
+#include <string>
+#include <vector>
+#include <memory>
+#include "core/framework/execution_providers.h"
+
+namespace onnxruntime {
+namespace lazytensor {
+
+class NvtxRange {
+ public:
+  NvtxRange(const char* name);
+  NvtxRange(const std::string& name);
+  ~NvtxRange();
+};
+
+// Class holding the CUDA EPs (one unique EP per device)
+// shared by all sessions.
+class CUDAExecutionProviderPool {
```

Comment:
there is a execution provider pool in training python module, will we merge these two part together in the future?

Reply:
Let me try to integrate them. If fail, I will create an issue for tracking. Btw, I have replaced LTC environment with ORT-eager's environment.